### ISSUE TITLE
Drag and drop to save file on the standard save file dialog

### ISSUE THREAD

**Issue by velissarious** [helloSystem/Filer#48]
In of the GUI of RISC OS when saving a file you can drag and drop the file before it has been created to the desired output directory using the mouse.

Look at this video of this feature:
https://youtu.be/oL4w3AK6Qpw?t=750

Some screenshots from the video:
The save dialog:
![image](https://user-images.githubusercontent.com/23398764/110257609-eba5bd00-7f9e-11eb-9cb7-866782ee522d.png)
Dragging the file to the output directory:
![image](https://user-images.githubusercontent.com/23398764/110257615-f6f8e880-7f9e-11eb-880d-b90c14f867d7.png)
Dropping it has placed it and saved in the desired directory.
![image](https://user-images.githubusercontent.com/23398764/110257620-024c1400-7f9f-11eb-9917-b37e5f5f2569.png)

**Comment by probonopd** [comment]
We want it to work the other way around: Drag a file from Filer into the Save dialog to save the file under the name of the existing file, thus overwriting it.

**Comment by linuxkettle** [comment]
I just take the first seven characters off.

**Comment by probonopd** [comment]
No, there must be a proper way to do this. I don't know why libfm-qt is using URIs internally in the first place. Probably a Glib/Gtk leftover.
